VM/bounding_box.py

```diff
@@ -0,0 +1,185 @@
+"""
+Module for handling bounding boxes in 2D space.
+
+This module provides classes and functions for working with bounding boxes in
+2D space, including calculating axis-aligned and minimum area bounding boxes,
+and computing various properties such as area and bearing. The bounding box
+dimensions are in metres except where otherwise mentioned.
+
+Classes:
+    - BoundingBox: Represents a 2D bounding box with properties and methods for calculations.
+
+Functions:
+    - axis_aligned_bounding_box: Returns an axis-aligned bounding box containing points.
+    - rotation_matrix: Returns the 2D rotation matrix for a given angle.
+    - minimum_area_bounding_box: Returns the smallest rectangle bounding points.
+    - minimum_area_bounding_box_for_polygons_masked: Returns a bounding box around masked polygons.
+"""
+
+import dataclasses
+
+import numpy as np
+import scipy as sp
+import shapely
+from qcore import geo
+from shapely import Polygon
+
+
+@dataclasses.dataclass
+class BoundingBox:
+    """Represents a 2D bounding box with properties and methods for calculations.
+
+    Attributes
+    ----------
+        corners : np.ndarray
+            The corners of the bounding box.
+    """
+
+    corners: np.ndarray
```

Comment:
Does the order of the corners in the array matter? I wonder if an example corners array or some ASCII art might be informative here

Reply:
It somewhat does, I can add a picture here.